client/src/store/reducer.ts

```diff
@@ -653,11 +677,13 @@ function onQuestionAnswered(
       status: MentorQuestionStatus.READY,
     };
 
-    const history = mentor.topic_questions.length - 1;
+    const history = mentor.mentor.topicQuestions.length - 1;
```

Comment:
it could break if the `mentor.mentor` does not exist, and it's also used below:
```jsx
!mentor.mentor.topicQuestions[history].questions.includes(
        response.question
      )
```
it would be good to check if `mentor.mentor` exists, then check if `mentor.mentor.topicQuestions` also exist

Reply:
Since these values are expected to be returned from the backend, it's actually beneficial for this portion of the code to break because then it tells us that there is an issue with the backend returning missing data. Although, there _should_ be an error boundary outside this function that will catch the error and handle it gracefully.

Reply:
great, so let's keep it!